service/dynamodb/api_client.go

```diff
@@ -78,6 +90,9 @@ type Options struct {
 	// Disabled by default.
 	EnableAcceptEncodingGzip bool
 
+	// Allows configuring endpoint discovery
+	EnableEndpointDiscovery aws.EndpointDiscoveryEnableState
```

Comment:
how is the endpoint for endpoint discovery option specified? looks like that option is only available for APIs that require endpoint discovery. Does it make sense to always have the property?

Reply:
We maintain a list of service ids in smithy codegen for which this option is available. This is not generated as per required/optional discovery. Only `timestream` services have the option for now. 

Adding services to the list is easy. I would like to keep it restricted until we have a feature request or requirement around it.